plugins/GSdx/GSDevice.cpp

```diff
@@ -133,9 +133,9 @@ void GSDevice::Present(GSTexture* sTex, GSTexture* dTex, const GSVector4& dRect,
 
 GSTexture* GSDevice::FetchSurface(int type, int w, int h, bool msaa, int format)
 {
-	GSVector2i size(w, h);
+	const GSVector2i size(w, h);
 
-	for(list<GSTexture*>::iterator i = m_pool.begin(); i != m_pool.end(); ++i)
+	for(std::deque<GSTexture*>::const_iterator i = m_pool.cbegin(); i != m_pool.cend(); ++i)
```

Comment:
AgePool can be slow. It is only called once by frame.

FetchSurface/Recycle are called more often. And I would say the same number of time. I don't think the performance is that critical for the code. However, if you don't find a `surface` in `FetchSurface`, you will call `CreateSurface` which is a tad slower (and requires a driver thread synchronization). Having too many texture in the pool and you will get an out-of-memory.

A standard PS2 texure is around `256x256 texels` (worst case is `1024x1024`). Smallest framebuffer is `1024 x 1024 x upscaling^2`